app/src/main/java/com/github/libretube/services/OnlinePlayerService.kt

```diff
@@ -138,9 +139,14 @@ open class OnlinePlayerService : AbstractPlayerService() {
             PlayingQueue.insertRelatedStreams(streams!!.relatedStreams)
         }
 
-        // save the current stream to the queue
         streams?.toStreamItem(videoId)?.let {
+            // save the current stream to the queue
             PlayingQueue.updateCurrent(it)
+
+            // update feed item with newer information, e.g. more up-to-date views
+            DatabaseHolder.Database.feedDao().update(
+                it.toFeedItem()
+            )
```

Comment:
I don't like to bypass the `FeedRepository` logic here, this should only be done if the `LocalFeedRepository` is used and thus we should likely add an abstract method `submitFeedItemChange` or something similar here:
https://github.com/libre-tube/LibreTube/blob/master/app/src/main/java/com/github/libretube/repo/FeedRepository.kt

I know that'd result in the same behavior because `@Update` wouldn't actually update anything if there's no local feed item, but that makes it easier to overview in the future.

Reply:
I'm personally not a fan of that abstraction, but I changed it.